packages/results/src/image.ts

```
@@ -0,0 +1,59 @@
+/**
+ * Material Design Icons as custom elements.
+ *
+ * @packageDocumentation
+ */
+
+import { GenericObject, Widget } from '@dfwc/core';
+import { property, CSSResultArray, html, customElement, css } from 'lit-element';
+import { styleMap } from 'lit-html/directives/style-map';
+
+import { styles } from './styles/image';
+
+/**
+ * Web component to display an image.
+ *
+ * @example
+ *
+ * ```html
+ * <df-image src="/images/logo.jpg"></df-icon>
+ * ```
+ *
+ * @public
+ */
+@customElement('df-image')
+export class Image extends Widget {
+  @property({ type: String }) public name = '';
+  @property({ type: String }) public src = '';
+  @property({ type: String }) public width = '';
+  @property({ type: String }) public height_ratio = '';
```

Use camel case for property names. You can set the attribute name in the property options:

```ts
@property({ type: String, attribute: 'height-ratio' }) public heightRatio;
```

Also, don't set empty strings as default values. If a required attribute is missing you can throw an exception or just render nothing (don't return anything in the render function).

---

The `fit` property (or `image-fit`) is missing, it should allow setting `object-fit` in the CSS.

```ts
export type ImageFit = 'contain' | 'fill' | 'cover' | 'none' | 'scale-down';
// …
@property({ type: String, attribute: 'image-fit' }) public fit: ImageFit = 'contain';
```

Take into account that you can set CSS based on attributes, see my comments in the styles file.